lib/aff4.py

```diff
@@ -2496,13 +2502,20 @@ def _ReadPartial(self, length):
 
     return result
 
-  def Read(self, length):
+  def Read(self, length=None):
     """Read a block of data from the file."""
     result = ""
 
     # The total available size in the file
-    length = int(length)
-    length = min(length, self.size - self.offset)
+    if length is not None:
+        length = int(length)
+        length = min(length, self.size - self.offset)
+    elif self.size < self.max_unbound_read:
```

Comment:
I think the difference is that your version accounts for cases where read(x) may have been called before read() and mine assumes only read() gets called so I was just raising on the file object size rather than amount to be read.  So the logic is correct, but yours is technically more correct so I can change it. :)